arch/tun_test.go

```diff
@@ -0,0 +1,14 @@
+package arch
+
+import "testing"
+
+func TestOpenTunDevice(t *testing.T) {
+	name := "tap0901"
+	addr := "192.168.0.2"
+	gw := "192.168.0.1"
+	mask := "255.255.255.0"
+	dnsServers := []string{"192.168.0.1"}
+	persist := false
+
+	OpenTunDevice(name, addr, gw, mask, dnsServers, persist)

```

Comment:
This only works on Windows and Mac but not Linux. Check the tun code for different platforms

Reply:
Yes, got it.